README.md

```diff
@@ -39,48 +39,84 @@ In particular, please read the [Pull Request Guidelines](https://github.com/exer
+enhancements be made with respect to the global problem specifications
+they may be performed by the presence of specifically named files in
+an exercises `.meta/` directory.
 
-* **leap.go** - This is a *stub file*, and will only be present in some exercises.
+So let's quickly run through each potential file and briefly describe
+what and why it is.
 
-* **leap_test.go** - This is the main test file for the exercise.
-
-* **.meta/** - The *.meta* directory contains files that are not meant to be
-    included when a user fetches an exercise, for instance test case generators.
-    See [ignored files](#ignored-files) for details on which files are ignored.
-
-* **gen.go** - This file, within the *.meta* directory, generates the *cases_test.go*
-    file, and will only be present in some exercises. See [generating test cases](#generating-test-cases)
-    for more information.
-
-In some exercises there can be extra files, for instance the `series` exercise
-contains extra test files.
+* **cases_test.go** - This file contains [generated test cases](#generating-test-cases),
+  and will only be present in some exercises. These will be 
+  automatically generated by a *.meta/gen.go* file and should
+  represent test data sourced from the problem-specifications repository.
+
+* **example.go** - This is an [example solution](#example-solutions) for
+  the exercise. It is used in integration testing and to provide a
+  reference example of a solution to the exercise. This file is ignored
+  by the `exercism fetch` command. See [ignored files](#ignored-files)
+  for details on which files are ignored.
+
+* **exercise_slug.go** - This is a *stub file*, and will only be
+  present in some exercises as a way to give users a useful starting
+  point.
+
+* **exercise_slug_test.go** - This is the main test file for the
+  exercise.
+
+* **.meta/** - The `.meta/` directory contains files that are not
+  to be included when a user fetches an exercise: test case generators
+  and README inclusion files. These files are for tooling and will not
+  be present for users of the exercise. (Again, see [ignored files](#ignored-files).)
+
+* **.meta/description.md** - This file, along with some others (below)
+  in the `.meta` directory are used to [generate exercise specific elements](https://github.com/exercism/docs/blob/master/language-tracks/exercises/anatomy/readmes.md)
+  of the exercise's README.md file. Any content in this file will
+  completely override the description.md of the exercise from the
+  [problem-specifications repository](https://github.com/exercism/problem-specifications/tree/master/exercises).
+
+* **.meta/gen.go** - This file, unique within the `.meta` directory,
+  generates the *cases_test.go* file, and will only be present in some
+  exercises should they have problem-specification test data and this
+  generator is present to to utilize it. See [generating test cases](#generating-test-cases) 
```

Comment:
This could be split for readability:

and will only be present in some
  exercises should they have problem-specification test data. This
  generator is present to to utilize it.